pkg/api/v1/types.go

```diff
@@ -944,6 +946,24 @@ const (
 	SecretVolumeSourceDefaultMode int32 = 0644
 )
 
+type SecretProjection struct {
+	// +optional
+	LocalObjectReference `json:"name,omitempty"`
```

Comment:
The tag for this should be: ``json:",inline"``.  Also, this is not an optional field.